src/View/Helper/PaginatorHelper.php

```diff
@@ -1153,26 +1153,40 @@ public function limitControl(array $limits = [], ?int $default = null, array $op
         $scope = $this->param('scope');
         assert($scope === null || is_string($scope));
 
-        $url = null;
         $currentPage = $this->paginated()->currentPage();
+        $query = $this->_View->getRequest()->getQueryParams();
 
-        if ($currentPage > 1) {
-            $query = $this->_View->getRequest()->getQueryParams();
-
-            if ($scope) {
-                $query[$scope]['page'] = 1;
-            } else {
-                $query['page'] = 1;
+        // Prepare query params to preserve as hidden fields
+        $hiddenFields = $query;
+        if ($scope) {
+            // Remove limit and page from scoped params
+            unset($hiddenFields[$scope]['limit'], $hiddenFields[$scope]['page']);
+            if (isset($hiddenFields[$scope]) && empty($hiddenFields[$scope])) {
+                unset($hiddenFields[$scope]);
+            }
+            // Set page to 1 if not on first page
+            if ($currentPage > 1) {
+                $hiddenFields[$scope]['page'] = 1;
+            }
+        } else {
+            // Remove pagination params that will be handled separately
+            unset($hiddenFields['limit'], $hiddenFields['page'], $hiddenFields['sort'], $hiddenFields['direction']);
+            // Set page to 1 if not on first page
+            if ($currentPage > 1) {
+                $hiddenFields['page'] = 1;
             }
-
-            $url = $this->_View->getRequest()->getPath();
-            $url .= '?' . http_build_query($query);
         }
 
         if ($scope) {
             $scope .= '.';
         }
+
+        $url = $this->_View->getRequest()->getPath();
         $out = $this->Form->create(null, ['type' => 'get', 'url' => $url]);
```

Comment:
The form URL defaults to current URL, so we can skip setting `$url`.

Reply:
If we remove the key, it generates the long ones again:

> /posts?filter%5Bdate%5D%5Bstart%5D=2024-01-01&amp;filter%5Bdate%5D%5Bend%5D=2024-12-31&amp;filter%5Bstatus%5D=active

which are somewhat overkill, since it skips the query string anyway.

Using `'url' => []` seems the best way, since it defaults to the current route URL.